etc/spack/defaults/modules.yaml

```diff
@@ -14,26 +14,35 @@
 #   ~/.spack/modules.yaml
 # -------------------------------------------------------------------------
 modules:
-  enable:
-    - tcl
-  prefix_inspections:
-    bin:
-      - PATH
-    man:
-      - MANPATH
-    share/man:
-      - MANPATH
-    share/aclocal:
-      - ACLOCAL_PATH
-    lib/pkgconfig:
-      - PKG_CONFIG_PATH
-    lib64/pkgconfig:
-      - PKG_CONFIG_PATH
-    share/pkgconfig:
-      - PKG_CONFIG_PATH
-    '':
-      - CMAKE_PREFIX_PATH
+  # These are configurations for the module set named "default"
+  default:
+    # Where to install modules
+    roots:
+      tcl:    $spack/share/spack/modules
+      lmod:   $spack/share/spack/lmod
+    # What time of modules to use
+    enable:
+      - tcl
```

Comment:
If we do this, we will have to either change our defaults or make it slightly more difficult for user to switch between module systems.

Currently, you don't need to specify a module root to change from tcl to lmod. If we mix the two settings, then someone who merely wants to change to lmod has to choose where those modules go, rather than getting a default from Spack.

This isn't insurmountable. We could simply require it, or we could change things so you specify a single root for both types. That said, I don't think it's worth pursuing as part of this PR at the moment unless you feel strongly otherwise.

Reply:
If we do this, we will have to either change our defaults or make it slightly more difficult for user to switch between module systems.

Currently, you don't need to specify a module root to change from tcl to lmod. If we mix the two settings, then someone who merely wants to change to lmod has to choose where those modules go, rather than getting a default from Spack.

This isn't insurmountable. We could simply require it, or we could change things so you specify a single root for both types. That said, I don't think it's worth pursuing as part of this PR at the moment unless you feel strongly otherwise.

Reply:
yeah let's keep it as is for now